kitsune/search/v2/documents.py

```diff
@@ -137,12 +137,37 @@ def get_model(cls):
         return wiki_models.Document
 
 
+class UserInnerDoc(InnerDoc):
+    id = field.Integer()
+    username = field.Keyword()
+
+
+class ProductInnerDoc(InnerDoc):
+    id = field.Integer()
+    title = field.Keyword()
+    slug = field.Keyword()
+
+
+class TopicInnerDoc(InnerDoc):
+    id = field.Integer()
+    title = field.Keyword()
+    slug = field.Keyword()
+
+
+class TagInnerDoc(InnerDoc):
+    id = field.Integer()
+    name = field.Keyword()
+    slug = field.Keyword()
```

Comment:
Currently tags will get flattened, the ES docs imply nested fields are quite expensive, and being able to filter on multiple fields of the same child object seem unnecessary here when id, name and slug could all be used equally effectively to filter.

Should we use an Object here at all, or should question_tags just be a list of tag slugs?